### PR TITLE
Add reads to test_simulated_stack.py

### PR DESCRIPTION
**Pull request by dcolligan**
Issue #462

### CODE REVIEW

File: tests/unit/test_simulated_stack.py [hunk 1]
```diff
@@ -155,3 +159,29 @@ def testCallSetsSearch(self):
...
+        self.assertEqual(response.status_code, 200)
+        responseData = protocol.SearchReadGroupSetsResponse.fromJsonString(
+            response.data)
+        readGroupSets = responseData.readGroupSets
+        self.assertEqual(len(readGroupSets), 1)
+
+        # search reads
+        path = utils.applyVersion('/reads/search')
+        request = protocol.SearchReadsRequest()
+        readGroupId = 'aReadGroupSet:one'
```
Comment (jeromekelleher) [inline tests/unit/test_simulated_stack.py]: This is pretty brittle --- why not get the readGroupId from the readGroupSets object you have above?

File: tests/unit/test_simulated_stack.py [hunk 2]
```diff
@@ -155,3 +159,29 @@ def testCallSetsSearch(self):
...
+        request.readGroupIds = [readGroupId]
+        response = self.sendJsonPostRequest(path, request.toJsonString())
+        self.assertEqual(response.status_code, 200)
+        responseData = protocol.SearchReadsResponse.fromJsonString(
+            response.data)
+        alignments = responseData.alignments
+        self.assertEqual(len(alignments), 2)
```
Comment (jeromekelleher) [inline tests/unit/test_simulated_stack.py]: How do you know this is 2?

**Comment by jeromekelleher** [comment]
I guess I'd like to see more automation here @dcolligan, and less coupling with the current state of the read simulator. However, the read simulator should be updated a lot too, as it's quite weak at the moment. We can open a separate issue on this if you prefer, and add in a requirement that this test code be generalised with that issue.

**Comment by dcolligan** [comment]
@jeromekelleher pushed a commit addressing your concerns

If you want a more comprehensive revamp of the simulated code, please file another issue specifying what you'd like to see.

**Comment by jeromekelleher** [comment]
Thanks @dcolligan --- a read simulator is a bigger task. This is good for now.

+1.

**Comment by dcolligan** [comment]
@jeromekelleher merge me!

**Pull request merged**